build/source/netcdf/modelwrite.f90

```diff
@@ -24,350 +24,314 @@ module modelwrite_module
+         case(iLookVarType%ifcToto); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcTotoStartIndex/),count=(/1,nLayers/))
+         case(iLookVarType%ifcSnow); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcSnowStartIndex/),count=(/1,nSnow/))
+         case(iLookVarType%ifcSoil); err = nf90_put_var(ncid(iFreq),meta(iVar)%ncVarID(iStat),(/dat(iVar)%dat/),start=(/iHRU,ifcSoilStartIndex/),count=(/1,nSoil/))
+        endselect ! vartype
+      endselect ! dat
+     endif ! sacalarv
+
+     ! process error code
+     if (err.ne.0) print*, meta(iVar)%ncVarID(iStat)
+     if (err.ne.0) message=trim(message)//trim(meta(iVar)%varName)//'_'//trim(get_statName(iStat))
+     call netcdf_err(err,message); if (err/=0) return
+
+    enddo ! iStat
+   enddo ! iVar
+  enddo ! iFreq
+
+ end subroutine writeData
+
+ ! **************************************************************************************
+ ! public subroutine writeBasin: write basin-average variables
+ ! **************************************************************************************
+ subroutine writeBasin(mstep,ostep,meta,stat,dat,indx,err,message)
+ USE data_types,only:var_info,dlength,ilength       ! type structures for passing
+ USE var_lookup,only:maxVarStat                     ! index into stats structure
+ USE var_lookup,only:iLookVarType                   ! index into type structure
+ USE var_lookup,only:iLookIndex                     ! index into index structure
+ USE var_lookup,only:iLookStat                      ! index into stat structure
+ USE globalData,only:outFreq,nFreq,ncid             ! output file information
+ USE get_ixName_module,only:get_varTypeName         ! to access type strings for error messages
+ USE get_ixName_module,only:get_statName            ! to access type strings for error messages
  implicit none
- ! input variables
- character(*), intent(in)      :: fileout        ! output file
- type(var_ilength),intent(in)  :: indx_data      ! model indices
- type(var_info),intent(in)     :: metaStruct(:)  ! metadata structure
- class(*),intent(in)           :: dataStruct     ! data structure
- integer(i4b), intent(in)      :: iHRU           ! hydrologic response unit
- integer(i4b), intent(in)      :: istep          ! model time step
- ! output variables
- integer(i4b),intent(out)      :: err            ! error code
- character(*),intent(out)      :: message        ! error message
+
+ ! declare dummy variables
+ type(var_info),intent(in)     :: meta(:)           ! meta data
+ type(dlength) ,intent(in)     :: stat(:)           ! stats data
+ type(dlength) ,intent(in)     :: dat(:)            ! timestep data
+ type(ilength) ,intent(in)     :: indx(:)           ! index data
+ integer(i4b)  ,intent(in)     :: mStep             ! model time step
```

Comment:
Changed all `mStep` and `oStep` variables to have the same names as in `multi_driver.f90` .